public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -98,16 +101,22 @@ export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDa
     if (obs.length === 1) {
       return obs[0];
     }
+
     if (obs.length > 1) {
-      // Not accurate, but simple and works
-      // should likely be more like the mixed data source
-      const promises = obs.map(o => o.toPromise());
-      return from(
-        Promise.all(promises).then(results => {
-          return { data: _.flatten(results) };
+      return forkJoin(obs).pipe(
+        map((results: DataQueryResponse[]) => {
+          const data: DataFrame[] = [];
+          for (const result of results) {
+            for (const frame of result.data) {
+              data.push(frame);
+            }
+          }
+
+          return { state: LoadingState.Done, data };
```

Comment:
The typings for this function `Observable<DataQueryResponseData>` seem incorrect when looking at what we are returning. But I just followed the return type that already existed in the code.

Reply:
weird -- it should be `Observable<DataDataQueryResponse>` I think, if that compiles and runs, we should change it

Reply:
Yes, I fixed it